src/glObjects.cpp

```diff
@@ -43,6 +54,22 @@ namespace gl
             glDisableVertexAttribArray(attrib);
     }
 
+    ScopedTexture::ScopedTexture(uint32_t target, uint32_t texture)
+        :target{ target }, texture{ texture }
+    {
+        // ES2 only supports 2D textures and cubemaps.
+        glGetIntegerv(target == GL_TEXTURE_2D ? GL_TEXTURE_BINDING_2D : GL_TEXTURE_BINDING_CUBE_MAP, &previously_bound);
```

Comment:
Most openGL guides say to avoid `glGet*` as they can be slow, no idea if this is the case for this one.

Reply:
That's not wrong, this is a safety net - I'm trying to minimize side effects.
On the same vein, you don't want to `Bind(GL_NONE)` all the time to "keep things clean" - it's a safety net as well.

So  far I've only seen gains  or no change (Oz on mac only reported a "more stable" fps) which means those few calls are not on the hot path (yet. When they are, that means the rendering is in an awesome shape 😁 )

Ideally you know your flow and you only rebind when necessary.
Which is something I want to get to... But first things first :)

Once all the 3D render code goes through `gl::Scoped*`, it becomes trivial to do client-side tracking.